lib/snapshot.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+const path = require('path');
+const fs = require('fs');
+const isEqual = require('lodash.isequal');
+const mkdirp = require('mkdirp');
+const globals = require('./globals');
+
+class Snapshot {
+	constructor(testPath, options) {
+		if (!testPath) {
+			throw new TypeError('Test file path is required');
+		}
+
+		this.dirPath = path.join(path.dirname(testPath), '__snapshots__');
+		this.filePath = path.join(this.dirPath, path.basename(testPath) + '.snap');
+		this.tests = {};
+		this.options = options || {};
+
+		if (fs.existsSync(this.filePath)) {
+			this.tests = JSON.parse(fs.readFileSync(this.filePath));
+		}
+	}
+
+	save() {
+		mkdirp.sync(this.dirPath);
```

Comment:
FYI, I forked off and tried this out and it works quite well: https://github.com/Omniroot/ava/pull/2/commits/c95962a4de900e7437965d5f7ecb04b6aaf98b9c